### PR TITLE
AzureAIsearchIndexDemo notebook updated to v0.10.3, AzureSearchvectorstore migrate to pydantic base and some others fixes

### PR DESCRIPTION
**Pull request by ryanrib14**
# Description
I'm just update the doc notebook of azureAisearch to work with new version v0.10.3 and i add some informations docstrings to my previously adjustments 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense
- [x] I ran the notebook code 

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by ryanrib14** [comment]
sorry, i 'm send the other past commit ( already merged) together, just the last real request review

File: docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb
Comment (logan-markewich) [inline docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb:1]: can we add to the pip install at the top? also need `pip install llama-index-vector-stores-azureaisearch`

Comment (ryanrib14) [reply docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb:1]: shure, i forgot this, sorry

Comment (ryanrib14) [reply docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb:1]: > can we add to the pip install at the top? also need `pip install llama-index-vector-stores-azureaisearch`

@logan-markewich everything right now, i guess

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py [hunk 1]
```diff
@@ -45,10 +46,27 @@ class IndexManagement(int, enum.Enum):
     CREATE_IF_NOT_EXISTS = auto()
 
 
-class AzureAISearchVectorStore(VectorStore):
+class AzureAISearchVectorStore(BasePydanticVectorStore):
     stores_text: bool = True
     flat_metadata: bool = True
 
+    from azure.search.documents.indexes import SearchIndexClient
+    from azure.search.documents import SearchClient
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:54]: These can just be imports at the top level I think

**Review by logan-markewich — APPROVED** [review]
Assuming you test this, it looks good to me 👍🏻

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py (hunk 1 above)
Comment (ryanrib14) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py]: ready!

**Comment by logan-markewich** [comment]
@ryanrib14 can you merge the latest main? It should help fix unit tests

**Comment by logan-markewich** [comment]
@ryanrib14 hmmm, interesting, unit tests are failing

**Comment by logan-markewich** [comment]
Need to update the test to use BasePydanticVectorStore

**Comment by ryanrib14** [comment]
> Need to update the test to use BasePydanticVectorStore

Sorry for the delay, it was a little late here in Brazil and I ended up not seeing it. Thanks for the fixes!

**Pull request merged**